pallets/task/src/lib.rs

```diff
@@ -400,10 +483,10 @@ pub mod pallet {
 
 		pub fn delete_task(task_initiator: &T::AccountId, task_id: &T::Hash) -> Result<(), DispatchError> {
 			// Check if task exists
-			let task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
+			let mut task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
 
```

Comment:
Only Completed tasks are allowed to be deleted

Reply:
Don't you call delete_task from on_initilize? 

If the deadline has passed, the task should be deleted if has not been started yet.

Reply:
Humm, needs some refactoring. Perhaps in next ticket.